### ISSUE TITLE
Add a postinstall hook for canisters

### ISSUE THREAD

**Issue by bitdivine** [dfinity/sdk#2387]
# Motivation
Some canisters have commands that need to be run after every installation.

Example: After installing the sns-wasm canister, it is necessary to upload artefacts to it.

# Proposal
Add "predeploy" and "postdeploy" entries, similar to the "build" command.

Like the build command, it would be either a string or an array of strings.

Each string would be a command, executed just before/after deployment.
```
{
  "dfx": "0.11.0",
  "canisters": {
    "wasm_canister": {
      "build": "curl --fail -sSL https://download.dfinity.systems/ic/268db05d70b3cc20475ba3ac046c0db9bd66c688/canisters/sns-wasm-canister.wasm.gz",
      "postdeploy": [ "./scripts/upload-wasms" ], <-- THIS IS THE POST INSTALLATION SETUP COMMAND
      "candid": "target/ic/sns_wasm.did",
      "wasm": "target/ic/sns-wasm-canister.wasm",
      "type": "custom"
    },
    "sn
```

**Comment by bitdivine** [comment]
If this is approved, in principle, I can submit a PR to implement it.

**Comment by adamspofford-dfinity** [comment]
Post-installation tasks were implemented in #2266. Pre-installation tasks are not necessary as you couldn't do anything useful for a Motoko canister and the other three types have built-in build scripts that pre-install tasks can be stuck in.
